server/src/main/java/org/elasticsearch/index/mapper/FieldMapper.java

```diff
@@ -1352,6 +1358,10 @@ public final void parse(String name, MappingParserContext parserContext, Map<Str
             validate();
         }
 
+        protected void handleUnknownParam(String propName, Object propNode) {
```

Comment:
should we rename it to something that removes any doubt that this is only called for legacy indices? handleLegacyindexUnknownParam ?

Reply:
I've renamed the method to handleUnknownParamOnLegacyIndex in f163412f3e6